include/deal.II/base/quadrature_point_data.h

```diff
@@ -732,18 +732,17 @@ pack_cell_data(const CellIteratorType &                           cell,
   const std::vector<std::shared_ptr<const DataType>> qpd =
     data_storage->get_data(cell);
 
-  const unsigned int n = matrix_data.n();
+  const auto m = qpd.size();
+  const auto n = qpd[0]->number_of_values();
```

Comment:
should assert that m>0

Reply:
...or change both variables `m` and `n` to be of type `const unsigned int`.

I would suggest to refrain from using `auto` here since `unsigned int` makes the code more easily readable.

Reply:
I believe that what @davydden is suggesting is that `qpd[0]` actually exists. So `Assert(m>0)` does seem appropriate here.